ros2_batch_job/packaging.py

```diff
@@ -112,30 +112,31 @@ def build_and_test_and_package(args, job):
     # Only on Linux and OSX Python scripts have a shebang line
     if args.os in ['linux', 'osx']:
         print('# BEGIN SUBSECTION: rewrite shebang lines')
-        bin_path = os.path.join(args.installspace, 'bin')
-        for filename in os.listdir(bin_path):
-            path = os.path.join(bin_path, filename)
-            with open(path, 'rb') as h:
-                content = h.read()
-            shebang = b'#!%b' % job.python.encode()
-            if content[0:len(shebang)] != shebang:
-                continue
-            print('- %s' % path)
-            with open(path, 'wb') as h:
-                h.write(b'#!/usr/bin/env python3')
-                h.write(content[len(shebang):])
+        # Demo nodes are installed to 'lib', so binaries may be present there in addition to 'bin'
+        for dir_with_binaries in ['bin', 'lib']:
+            path_with_binaries = os.path.join(args.installspace, dir_with_binaries)
+            for path in [os.path.join(root, f) for root, dirs, files in os.walk(path_with_binaries) for f in files]:
```

Comment:
How it is at the moment, the `bin` and `lib` directories are both being walked recursively with no limit. Could you explain the harm of considering the parent `lib` folder and the subdirectories of `bin`? We don't install anything to either at the moment, but if we were to we would also want the shebang to be re-written I would think.

This can be refactored to act differently for `lib` and `bin` but I would like to understand why it's necessary.